libs/utils/src/filter.c

```diff
@@ -440,14 +516,27 @@ static bool celix_filter_hasFilterChildren(celix_filter_t* filter) {
  */
 static celix_status_t celix_filter_compile(celix_filter_t* filter) {
     if (celix_filter_isCompareOperand(filter->operand)) {
-        filter->internal = malloc(sizeof(*filter->internal));
+        filter->internal = calloc(1, sizeof(*filter->internal));
         if (filter->internal == NULL) {
+            celix_err_push("Filter Error: Failed to allocate memory.");
             return CELIX_ENOMEM;
-        } else {
-            filter->internal->longValue = celix_utils_convertStringToLong(filter->value, 0, &filter->internal->convertedToLong);
-            filter->internal->doubleValue = celix_utils_convertStringToDouble(filter->value, 0.0, &filter->internal->convertedToDouble);
-            filter->internal->versionValue = celix_utils_convertStringToVersion(filter->value, NULL, &filter->internal->convertedToVersion);
         }
+        do {
+            filter->internal->longValue =
+                    celix_utils_convertStringToLong(filter->value, 0, &filter->internal->convertedToLong);
+            filter->internal->doubleValue =
+                    celix_utils_convertStringToDouble(filter->value, 0.0, &filter->internal->convertedToDouble);
+            if (filter->internal->convertedToLong || filter->internal->convertedToDouble) {
+                break;
+            }
+            filter->internal->boolValue =
+                    celix_utils_convertStringToBool(filter->value, false, &filter->internal->convertedToBool);
```

Comment:
For filter with long string value, the original implementation will do dynamic memory allocation, which is eliminated by the current implementation of `celix_utils_convertStringToBool`.